### PR TITLE
Route building from string interpolation to path.join

### LINKED ISSUE #1284: Modify adminRoutes and talentSearchRoutes so APP_DIR can start with "/"
# Description
Currently, for talentsearchRoutes to work correctly, TALENTSEARCH_APP_DIR must be the empty string. This is inconvenient on the server infrastructure. Adjust the ____Routes.ts files so that expect ___APP_DIR to start with a backslash. Or, even better, allow them to work either way (starting with a backslash or not) and also work if APP_DIR is "/".

### PR DESCRIPTION
**Pull request by petertgiles**
This branch changes route building from string interpolation to using [Node's path.join function](https://nodejs.org/api/path.html#pathjoinpaths).  This allows the route building functions to work properly regardless of whether the base directory starts with a slash or not.

A dependency on path-browserify is added.

Closes #1284

### CODE REVIEW

File: admin/resources/js/adminRoutes.ts [hunk 1]
```diff
@@ -1,57 +1,71 @@
+import path from "path-browserify";
 import { ADMIN_APP_DIR } from "./adminConstants";
 
-export const homePath = (): string => `/${ADMIN_APP_DIR}`;
+export const homePath = (): string => path.join("/", ADMIN_APP_DIR); // leading slash in case empty base url
 export const homePublicPath = (): string =>
-  `/${ADMIN_APP_DIR}/public/${ADMIN_APP_DIR}`;
```
Comment (petertgiles) [inline admin/resources/js/adminRoutes.ts:5]: I feel this original line with `ADMIN_APP_DIR` twice was a bug.  Was I correct to remove the second instance of it?

Comment (tristan-orourke) [reply admin/resources/js/adminRoutes.ts:5]: I think it was more of a workaround than a bug - our htaccess used to redirect you to /admin/public/admin sometimes. Happily, we've resolved those issues. We should be able to remove `homePublicPath` entirely, actually.

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Nice, use of a path function makes sense here. I see that in adminRoutes we're referencing ADMIN_APP_DIR in each function, whereas in talentSearchRoutes we prefix subsequent urls with `homePath()`. I think either approach is acceptable, but this seems like a opportunity to make them consistent.

File: admin/resources/js/adminRoutes.ts (hunk 1 above)
Comment (petertgiles) [reply admin/resources/js/adminRoutes.ts:5]: Ah, that makes sense.  OK, I've gone ahead and removed the constant entirely and routing seems to still work fine on the admin site.

**Comment by petertgiles** [comment]
> this seems like a opportunity to make them consistent

Good call on this.  I prefer the way it is on the talentsearch site so I've updated the admin side to match.

**Review by tristan-orourke — APPROVED** [review]

**Pull request merged**